src/nvim/option.c

```diff
@@ -196,7 +196,7 @@ typedef struct vimoption {
   idopt_T indir;                /* global option: PV_NONE;
                                  * local option: indirect option index */
   char_u      *def_val[2];      /* default values for variable (vi and vim) */
-  scid_T scriptID;              /* script in which the option was last set */
+  LastSet last_set;             /* script in which the option was last set */
```

Comment:
I would add a new field `vimoption_T.chan_id` rather than defining a `LastSet` type.

Reply:
`LastSet` is also used in `buffer_defs.h` for `winopt_T` and `file_buffer`. Should I add the field `int chan_id[]` to those structs as well?

Reply:
Right again, your way is better. Cancel my suggestion.